pkg/controller/pxc/full_crash_recovery.go

```diff
@@ -0,0 +1,153 @@
+package pxc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	v1 "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	ErrNotAllPXCPodsRunning = errors.New("not all pxc pods are running")
+	logLinesRequired        = int64(8)
+)
+
+const logPrefix = `#####################################################LAST_LINE`
+
+func (r *ReconcilePerconaXtraDBCluster) recoverFullClusterCrashIfNeeded(cr *v1.PerconaXtraDBCluster) error {
+	if cr.Spec.PXC.Size <= 0 {
+		return nil
+	}
+
+	err := r.checkIfPodsRunning(cr)
+	if err != nil {
+		if err == ErrNotAllPXCPodsRunning {
+			return nil
+		}
+		return err
+	}
+
+	logOpts := &corev1.PodLogOptions{
+		Container: "pxc",
+		TailLines: &logLinesRequired,
+	}
+	logLines, err := r.clientcmd.PodLogs(cr.Namespace, cr.Name+"-pxc-0", logOpts)
+	if err != nil {
+		return errors.Wrap(err, "get logs from pxc 0 pod")
+	}
+
+	for _, log := range logLines {
+		if strings.HasPrefix(log, logPrefix) {
+			return r.doFullCrashRecovery(cr.Name, cr.Namespace, int(cr.Spec.PXC.Size))
+		}
+	}
```

Comment:
Just call `isPodWaitingForRecovery()` here to define if doFullCrashRecovery needed. No sense to duplicate this code

Reply:
fixed there
9452638